### ISSUE TITLE
Sharing Links with password can be accessed without password

### ISSUE THREAD

**Issue by Crashman1983** [bpatrik/pigallery2#950]
**Describe the bug**

I am logged in with my personal account, which is admin.
I want to share a directory with several subdirectories. Therefore I created a share link for the top level directory. I limited the link to 30d and set a password. 

When I open the newly created share link in a private browser window or on a different client (smartphone) it will open the link and show the gallery without asking for password. The user shown is "guest" which was not configured before. 

I cannot protect a share link with a password. I would expect PiGallery2 to ask for a password before showing the directory content.

**Environment (please complete the following information):**
 - OS: linux, docker
 - Browser: safari, edge

**Used app version**:
- release v2.0 (docker)

**Comment by israel-lugo** [comment]
Is there any news on this? I also have observed the same issue and can't trust sharing because of it. This is a serious security/privacy problem.

I created a share with a password. Opening the share link on a private browser window takes me straight to the photos, with a "Guest" user. No prompt for password.

If I hit Back a couple of times eventually I do get thrown into a password prompt, but by then it's too late.

Also running v2.0.0 on Docker.

**Comment by israel-lugo** [comment]
Tagging @bpatrik for awareness

**Comment by PawelWos1987** [comment]
Same problem here. Qnap Container station.

**Comment by bpatrik** [comment]
Related to #744. Although, I hopped that its already fixed.

**Comment by bpatrik** [comment]
Yeah it is fixed since dec 23 in [4d7e7fe](https://github.com/bpatrik/pigallery2/commit/4d7e7fe1613c1695c830c66b0edbcffa28b11117), [16bf756](https://github.com/bpatrik/pigallery2/commit/16bf7565820e4bb528f76e5b3725982a4c6b64d4) and [ba9b529](https://github.com/Rickkwa/pigallery2/commit/ba9b5292e1f5d7559df078a855258a6b23d4c797)

please use `:edge` or wait for the next release. I don't plan to add much more features to edge, only to fix bugs before the next release.
